docs/DeveloperGuide.md

```diff
@@ -383,6 +383,45 @@ The sequence diagram is as follows:
 
 The exported file can be subsequently imported back into any other instance of ContaX, similar to the existing `.json` system of import/export.
 
+### Enhanced Find Logic
+
+The previous implementation find target is limited to name field only. Hence, extending the find function will increase the user experience by enable them easily search `Person` with `phone`
+`email` and `address` field.
+
+This is done by creating `searchType` model within the `model util` package and extracting the original `NameConstrainsKeywordsPreicate` to form `ConstrainsKeywordsPreicate` then create different inherit of three different types of predicate.
+
+The sequence diagram is as follows:
+![Enhanced Find Logic](images/FindCommandSequenceDiagram.png)
+
+### Enhanced Command Features
+
+#### Chain Command
+
+This extension allows user to chain multiple commands together.
+
+:information_source: **Note:** A command failure may lead to following commands being invalid (e.g. if create person failed, you cannot edit the newly created person), causing an exception to be thrown.
+
+The sequence diagram is as follows:
+![Chain Command](images/ChainCommandSequenceDiagram.png)
+
+#### Range Command
+
+This extension allows user to perform range of commands base on `index`. Hence, `from/INDEX to/INDEX` is essential in parsing the value of input. The validation of index is similar to original command (i.e. the index in edit command).
+
+#### Batch Command
+
+This extension focuses on matching specific value field (`name phone email address`) and then perform edit or delete operation on the target which satisfy the matching condition. This is achieved by `BatchCommand#matchInputStringToIndex()` which converts the person that matched the condition and returned as series of command and execute sequentially.
```

Comment:
What do you mean by value fields?
```suggestion
This extension allows the editing of `Person` objects that have attributes matching a specific value. Since this matching requires the objects to already exist, only edit and delete operations can be performed. The command translates the `Person` objects matching the condition into a series of indexes and executes the specified command on them sequentially.
```

Reply:
updated accordingly